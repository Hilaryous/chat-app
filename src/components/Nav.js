// @flow
import React from 'react'
import styled from 'react-emotion'
import AppContext from '../context'
import Button from './Button'

type Props = {
  email: string,
  logoutUser: () => void,
}

const Nav = ({ email, logoutUser }: Props) => (
  <Relative>
    <Flex
      alignItems="center"
      height="100%"
      justifyContent="space-between"
      px={3}
    >
      <p>
        {`Hi! ${email}`}
      </p>
      <Button
        onClick={logoutUser}
      >
        Logout
      </Button>
    </Flex>
  </Relative>
)

export default () => (
  <AppContext.Consumer>
    {({ currentUser, actions: { logoutUser } }) => (
      <Nav
        logoutUser={logoutUser}
        email={currentUser.email}
      />
    )}
  </AppContext.Consumer>
)

//------------------------------

const Relative = styled('div')({
  width: '100%',
  height: 80,
  borderBottom: '2px black solid',
})

const Flex = styled('div')({
  marginLeft: 'auto',
  marginRight: 'auto',
  paddingLeft: '16px',
  paddingRight: '16px',
  maxWidth: '1440',
  display: 'flex',
  alignItems: 'center',
  height: '100%',
  justifyContent: 'space-between',
  px: '16px',
})
