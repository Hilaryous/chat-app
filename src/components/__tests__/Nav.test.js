import React from 'react'
import renderer from 'react-test-renderer'
import { StaticRouter } from 'react-router-dom'
import Nav from '../Nav'
import AppContext from '../../context'

describe('Nav', () => {
  it('shows the correct snapshot tree for the Nav component', () => {
    const tree = renderer.create(
      <StaticRouter location="someLocation" context={{}}>

        <AppContext.Provider>
          <Nav />
        </AppContext.Provider>
      </StaticRouter>,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
