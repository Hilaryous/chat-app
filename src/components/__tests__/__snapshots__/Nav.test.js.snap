// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav shows the correct snapshot tree for the Nav component 1`] = `
<div
  className="css-1b2b698"
>
  <div
    className="css-1bbbaa"
    height="100%"
  >
    <p>
      Hi! 
    </p>
    <button
      className="css-bjolns"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</div>
`;
